src/XRP/issued-currency-client.ts

```diff
@@ -372,4 +373,59 @@ export default class IssuedCurrencyClient {
       wallet,
     )
   }
+
+  public async authorizeTrustline(
+    accountToTrust: string,
+    currencyName: string,
+    wallet: Wallet,
+  ): Promise<TransactionResult> {
+    if (!XrpUtils.isValidXAddress(accountToTrust)) {
+      throw XrpError.xAddressRequired
+    }
+    const classicAddress = XrpUtils.decodeXAddress(accountToTrust)
+    if (!classicAddress) {
+      throw XrpError.xAddressRequired
+    }
+
+    // TODO (tedkalaw): Remove this when ripple-binary-codec supports X-Addresses.
+    const issuerAccountAddress = new AccountAddress()
+    issuerAccountAddress.setAddress(classicAddress.address)
+
+    const currency = new Currency()
+    currency.setName(currencyName)
+
+    const issuedCurrencyAmount = new IssuedCurrencyAmount()
+    issuedCurrencyAmount.setCurrency(currency)
+    issuedCurrencyAmount.setIssuer(issuerAccountAddress)
+
+    // When authorizing a trustline, set it to 0 first.
+    // TODO: Include doc.
+    issuedCurrencyAmount.setValue('0')
+
+    const currencyAmount = new CurrencyAmount()
+    currencyAmount.setIssuedCurrencyAmount(issuedCurrencyAmount)
+
+    const limit = new LimitAmount()
+    limit.setValue(currencyAmount)
+
+    const trustSet = new TrustSet()
+    trustSet.setLimitAmount(limit)
+
+    const transaction = await this.coreXrplClient.prepareBaseTransaction(wallet)
+    transaction.setTrustSet(trustSet)
+
+    const flags = new Flags()
+    flags.setValue(TrustSetFlag.tfSetfAuth)
+    transaction.setFlags(flags)
```

Comment:
@amiecorso am i doing this right? per my huge slack message i sent you, this does not appear to end up in the transaction. is it possible we aren't actually serializing flags?

https://testnet.xrpl.org/transactions/CCAA5AB0F281CA716FF153140B3E514FC7D77640776884ADD6218CA41EA8802F/raw